### PR TITLE
JENKINS-38975 Add rebuild on pull request comment

### PR DESCRIPTION
**Pull request by bluesliverx**

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubPullRequestCommentCause.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016 CloudBees, Inc.
```
Comment (orrc) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubPullRequestCommentCause.java]: Do you work at CloudBees? :)

File: src/main/java/org/jenkinsci/plugins/github_branch_source/IssueCommentGHEventSubscriber.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            public void run() {
+                boolean jobFound = false;
+                for (final SCMSourceOwner owner : SCMSourceOwners.all()) {
+                    for (SCMSource source : owner.getSCMSources()) {
+                        if (!(source instanceof GitHubSCMSource)) {
+                            continue;
+                        }
+                        GitHubSCMSource gitHubSCMSource = (GitHubSCMSource) source;
+                        if (gitHubSCMSource.getRepoOwner().equals(changedRepository.getUserName()) &&
+                                gitHubSCMSource.getRepository().equals(changedRepository.getRepositoryName())) {
```
Comment (orrc) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/IssueCommentGHEventSubscriber.java]: Be careful with case-sensitive comparisons: [JENKINS-39114](https://issues.jenkins-ci.org/browse/JENKINS-39114)

Comment (bluesliverx) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/IssueCommentGHEventSubscriber.java]: Fixed

**Comment by bluesliverx** [comment]
Closing this as I'm going to create a separate plugin for this, waiting months is not acceptable to us at the moment and it doesn't directly modify any existing code.

**Comment by bluesliverx** [comment]
https://github.com/bluesliverx/github-pr-comment-build-plugin

**Comment by KostyaSha** [comment]
@bluesliverx hi, i have in plans to implement multibranch in https://github.com/KostyaSha/github-integration-plugin that will have all this features. Current branch-source plugin already reinvented most of features that existed for year.

**Comment by oleg-nenashev** [comment]
@bluesliverx FYI you can mention @jenkinsci/code-reviewers if you feel the review process is stuck.

**Comment by bluesliverx** [comment]
Thanks @oleg-nenashev, good info for the future!

File: src/main/java/org/jenkinsci/plugins/github_branch_source/TriggerPRCommentBranchProperty.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import hudson.model.Run;
+import jenkins.branch.BranchProperty;
+import jenkins.branch.BranchPropertyDescriptor;
+import jenkins.branch.JobDecorator;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Allows a GitHub pull request comment to trigger an immediate build based on a comment string.
+ */
+public class TriggerPRCommentBranchProperty extends BranchProperty {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/TriggerPRCommentBranchProperty.java]: `BranchProperty` will not suffice for a GH org folder I think. You would need an `AbstractFolderProperty<OrganizationFolder>`.

**Pull request closed without merging**